tests/functional/test_basic.py

```diff
@@ -417,6 +417,29 @@ def test_time_module_localized(pyi_builder, monkeypatch):
         """)
 
 
+# Check that `locale.getlocale()` in frozen application returns user-preferred locale (i.e., that user-preferred
+# locale is set in the bootloader during python interpreter setup). In its current form, the test is mostly ensuring
+# that the user-preferred locale is set *at all* (see #8305).
+#
+# NOTE: monkey-patching LC_ALL might cause issues during build. Ideally, we would modify the LC_ALL only when running
+# built executable, but currently `pyi_builder.test_source` does not allow propagation of run-time environment
+# variables. If that is ever fixed/extended, the test could include a wider range of locale combinations to test with
+# (for example, sl_SI.UTF-8, sl_SI.ISO8859-2).
```

Comment:
Hmm, true. We could pass the expected locale and encoding via command-line arguments, and then use a single build to test multiple variants. Without arguments, it would just check that values are not `None`, and with arguments, it would check that they are what they expect them to be.